# 25 · GSE135779 · scRNA_seq · WGCNA modules

Reads `wgcna_input.rds`. Writes `modules.rds` and the end product `module_genes.csv` in
`data/run_artifacts/GSE135779/`.

Settings as for GSE65391 (notebook 05): signed network and TOM, minModuleSize 30, mergeCutHeight 0.25.
There are more genes than fit one block, so `maxBlockSize = 5000`: WGCNA pre-clusters the genes into
blocks and builds modules within each block.

## Load WGCNA and the input from notebook 24

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
w <- readRDS(art("GSE135779", "wgcna_input.rds"))
c(samples = nrow(w$datExpr), genes = ncol(w$datExpr), power = w$power)

samples   genes   power 
     33   15353      18

**Explanation**
- As in notebook 05, with the input of notebook 24.

## Find the modules

In [2]:
set.seed(SEED)
net <- blockwiseModules(w$datExpr, power = w$power, networkType = "signed", TOMType = "signed",
                        minModuleSize = 30, mergeCutHeight = 0.25, maxBlockSize = 5000,
                        numericLabels = FALSE, pamRespectsDendro = FALSE, verbose = 0)
sizes <- sort(table(net$colors), decreasing = TRUE)
c(modules = sum(names(sizes) != "grey"), genes_in_modules = sum(sizes[names(sizes) != "grey"]), grey = unname(sizes["grey"]))
sizes

modules genes_in_modules             grey 
              27            14070             1283


    turquoise          blue         brown          grey        yellow 
         1845          1802          1540          1283           891 
        green           red         black          pink       magenta 
          847           738           733           629           623 
       purple   greenyellow           tan        salmon          cyan 
          546           545           487           347           330 
 midnightblue     lightcyan        grey60    lightgreen   lightyellow 
          328           260           246           238           223 
    royalblue       darkred     darkgreen darkturquoise      darkgrey 
          195           172           155           131            73 
       orange    darkorange         white 
           54            47            45 

**Explanation**
- As in notebook 14, including `maxBlockSize = 5000`.

**Result.** 30 modules holding 13,943 of 15,353 genes; 1,410 are grey.

## Module membership and hub genes

In [3]:
ME  <- orderMEs(net$MEs)
kME <- cor(w$datExpr, ME)
modules <- setdiff(names(sizes), "grey")
hubs <- lapply(setNames(modules, modules), function(mo) {
  g <- names(net$colors)[net$colors == mo]
  names(sort(kME[g, paste0("ME", mo)], decreasing = TRUE))
})
t(sapply(hubs, function(h) paste(head(h, 8), collapse = ", ")))

turquoise,blue,brown,yellow,green,red,black,pink,magenta,purple,⋯,lightgreen,lightyellow,royalblue,darkred,darkgreen,darkturquoise,darkgrey,orange,darkorange,white
"EMILIN2, CTNNA1, CYBB, ATP11A, CORO1C, MFSD1, LPCAT2, LACTB","RPSA, RPL19, FUS, RPL29, EPB41L4A-AS1, RPS20, HNRNPA1, TOMM20","CAMK4, EVL, DNMT3A, CHD3, ABHD14B, DGKA, RNF216, CD3E","KBTBD2, FOSL2, GABARAPL1, YTHDF3, MNT, ZRANB1, MIDN, JUND","TMEM259, RASAL3, SUN2, P2RY8, AKAP8L, EML3, DVL2, HSF1","DNAJA3, ZNF146, PYGO2, JPT2, THNSL1, ZNF444, NHLRC4, OLFM2","TRMT13, NIFK-AS1, LYRM7, TUBD1, SCAI, SYNJ2BP, MTMR4, TRAF3IP3","RPS6KA4, ATG16L2, UBXN11, ARAP1, MAP3K11, PLCB3, PEA15, ROGDI","CCNB1, CDCA8, RAD51, HJURP, ASF1B, KIFC1, BUB1, CCNA2","GEMIN6, RPP25L, TMEM223, BET1, ANAPC10, NDUFA8, EXOSC3, MRPL35",⋯,"MAPK1IP1L, PPP2CA, SLC71A1, EAPP, BZW1, ARHGDIA, DNAJB6, CNOT1","CTD-2192J16.22, RP11-770J1.5, MTG1, XXbac-BPG252P9.9, RP11-138I1.4, NME2, RP11-498C9.2, SPDYE2","PLCG2, CD180, ATP13A1, LINC03034, TLR7, E2F5, DHTKD1, RPP25","CCL5, CST7, PPP2R2B, FCRL6, C1orf21, TGFBR3, GZMH, KIF21A","ATIC, ZNF138, CCT7, TMEM263, ALDOC, DDX50, PRMT1, VRK3","CMTM5, TUBB1, ACRBP, SPARC, CTTN, CAVIN2, GP9, GFI1B","ARRB2, SPG21, COTL1, TAF6L, HOTAIRM1, FCGRT, ATP6AP2, DAZAP2","SLC4A1, SNCA, CA1, LINC00570, SELENBP1, TMC5-AS1, ALAS2, AHSP","FKBP5, CELF2, SMAP2, TXNIP, SLA, SLC35E3, SESN1, CCND3","NUDT16, TMIGD3, SRGAP1, CD163, SAP30, ALOX15B, SLC1A3, JDP2"


**Explanation**
- As in notebook 05.

## Which modules hold the interferon genes?

In [4]:
ifn6 <- read_gene_set("ifn-type1-6.txt")
data.frame(gene = ifn6, module = net$colors[ifn6], row.names = NULL)

gene,module
<chr>,<chr>
IFI27,lightcyan
IFI44L,lightcyan
IFIT1,lightcyan
ISG15,lightcyan
RSAD2,lightcyan
SIGLEC1,lightcyan


**Explanation**
- As in notebook 05.

**Result.** All six interferon-score genes are in one module, magenta (hub genes IFI35, HESX1, FBXO6, LAP3,
STAT2). Other modules read by their hub genes: red (CCNB1, RAD51, CDK1: cell cycle), salmon (FCRLA, MS4A1,
CD19, CD79B: B cell), grey60 (CCL5, CST7, FCRL6: NK / cytotoxic), darkgreen (CMTM5, TUBB1, SPARC:
platelet), skyblue (CA1, SLC4A1, SELENBP1: red cell) and turquoise (BCL11B, RPSA: lymphoid and ribosomal).

## Save the modules

In [5]:
in_module <- names(net$colors)[net$colors != "grey"]
module_genes <- data.frame(gene = in_module, module = net$colors[in_module],
                           kME = kME[cbind(in_module, paste0("ME", net$colors[in_module]))], row.names = NULL)
write.csv(module_genes, art("GSE135779", "module_genes.csv"), row.names = FALSE)
saveRDS(list(colors = net$colors, MEs = ME, kME = kME, hubs = hubs), art("GSE135779", "modules.rds"))

**Explanation**
- As in notebook 05. These are this study's own modules, not the array modules.